src/game/objects/har.c

```diff
@@ -1758,6 +1758,63 @@ static void har_palette_transform(damage_tracker *damage, vga_palette *pal, void
     }
 }
 
+void har_handle_stun(object *obj) {
+    har *h = object_get_userdata(obj);
+
+    if(h->health <= 0) {
+        h->endurance = h->endurance_max - 2;
+    } else {
+        if(h->endurance < 1) { // We are currently dizzy, recover gradually
+            if(obj->cur_animation->id == ANIM_STUNNED) {
+                h->endurance += h->stun_factor * 1.4;
+            }
+        } else { // Not yet stunned, handle regular stun recovery
+            if((obj->cur_animation->id == ANIM_IDLE) || (obj->cur_animation->id == ANIM_CROUCHING) ||
+               (obj->cur_animation->id == ANIM_VICTORY)) {
+                float hpfactor = ((h->health_max * 5.0 / 9.0) + h->health) / h->health_max;
+                float stunfactor = 1.0 * h->stun_factor * h->endurance / h->endurance_max / 256.0;
```

Comment:
Stunfactor seems to always be the same in all these cases, so just define it once?